qiskit_dynamics/backend/dynamics_backend.py

```diff
@@ -351,6 +374,159 @@ def target(self) -> Target:
+            :class:`.DynamicsBackend`.
+
+        The ``backend`` must have the ``configuration`` and ``defaults`` attributes. The
+        ``configuration`` must containing a Hamiltonian description, step size ``dt``, number of
+        qubits ``n_qubits``, and ``u_channel_lo`` (when control channels are present).  The
+        ``defaults`` must contain ``qubit_freq_est``, as well as ``meas_freq_est`` if measurement
+        channels appear explicitly in the Hamiltonian.
+
+        .. note::
+
+            The ``configuration`` and ``defaults`` attributes of the original backend are not copied
+            into the constructed :class:`DynamicsBackend` instance, only the required data stored
+            within these attributes will be extracted. If required, for backwards compatibility,
+            ``'configuration'`` and ``'defaults'`` options can be set, which will be returned via
+            the :meth:`.configuration` and :meth:`.defaults` methods.
+
+        The optional argument ``subsystem_list`` specifies which subset of qubits to model in the
+        constructed :class:`DynamicsBackend`. All other qubits are dropped from the model.
+
+        Configuration of the underlying :class:`.Solver` is controlled via the ``rotating_frame``,
+        ``evaluation_mode``, and ``rwa_cutoff_freq`` options. In contrast to :class:`.Solver`
+        initialization, ``rotating_frame`` defaults to the string ``"auto"``, which allows this
+        method to choose the rotating frame based on ``evaluation_mode``:
+
+        * If a dense evaluation mode is chosen, the rotating frame will be set to the
+          ``static_hamiltonian`` indicated by the Hamiltonian in ``backend.configuration()``.
+        * If a sparse evaluation mode is chosen, the rotating frame will be set to the diagonal of
+          ``static_hamiltonian``.
+
+        Otherwise the ``rotating_frame``, ``evaluation_mode``, and ``rwa_cutoff_freq`` are passed
+        directly to the :class:`.Solver` initialization.
+
+        Args:
+            backend: The ``Backend`` instance to build the :class:`.DynamicsBackend` from.
+            subsystem_list: The list of qubits in the backend to include in the model.
+            rotating_frame: Rotating frame argument for the internal :class:`.Solver`. Defaults to
+                ``"auto"``, allowing this method to pick a rotating frame.
+            evaluation_mode: Evaluation mode argument for the internal :class:`.Solver`.
+            rwa_cutoff_freq: Rotating wave approximation argument for the internal :class:`.Solver`.
+            **options: Additional options to be applied in construction of the
+                :class:`.DynamicsBackend`.
+
+        Returns:
+            DynamicsBackend
+
+        Raises:
+            QiskitError: If any required parameters are missing from the passed backend.
+        """
+
+        if not hasattr(backend, "configuration"):
```

Comment:
Ok, I guess this is where it comes from. What about for V2 backends from IBM provider, shouldn't these be able to get all this info from target? (@mtreinish might have advice)

Reply:
For the IBM provider's V2 backends the `configuration()`, `properties()`, and `defaults()` methods still exist as they're custom parts of the ibm provider's implementation of `BackendV2`.  We're safe to do this even with `BackendV2` if we're ok saying that this method will only work with the IBM provider. We can't do this for any other `BackendV2` as these methods are explicitly not part of the base API.